runtime/src/main/java/org/corfudb/runtime/view/QuorumReplicationView.java

```
@@ -0,0 +1,117 @@
+                     new AutoCloseableByteBuf(ByteBufAllocator.DEFAULT.directBuffer())) {
+            Serializers.CORFU.serialize(data, b);
+
+            LogData ld = new LogData(DataType.DATA, b);
+            ld.setBackpointerMap(backpointerMap);
+            ld.setStreams(stream);
+            ld.setGlobalAddress(address);
+
+            // FIXME
+            if (data instanceof LogEntry) {
+                ((LogEntry) data).setRuntime(getLayout().getRuntime());
+                ((LogEntry) data).setEntry(ld);
+            }
+
+            payloadBytes = b.readableBytes();
+            CompletableFuture<Boolean>[] futures = new CompletableFuture[numUnits];
+            for (int i = 0; i < numUnits; i++) {
+                log.trace("Write[{}]: quorum {}/{}", address, i + 1, numUnits);
+                // sending the message to all logging units and ignoring the exception.
+                futures[i] = getLayout().getLogUnitClient(address, i).write(address, stream, 0L, data, backpointerMap);
+            }
+            Boolean result = CFUtils.getUninterruptibly(QuorumFutureFactory.getQuorumFuture(futures));
+            log.trace("Write[{}]: {}", address, result);
+        }
+        return payloadBytes;
+    }
+
+    /**
+     * Read the given object from an address, using the replication method given.
+     *
+     * @param address The address to read from.
+     * @return The result of the read.
+     */
+    @Override
+    public LogData read(long address) {
+        int numUnits = getLayout().getSegmentLength(address);
+        log.trace("Read[{}]: chain {}/{}", address, numUnits, numUnits);
+        CompletableFuture<ReadResponse>[] futures = new CompletableFuture[numUnits];
+            for (int i=0; i<numUnits; i++) {
+                futures[i]=getLayout().getLogUnitClient(address, i).read(address);
+        }
+        ReadResponse readResponse = CFUtils.getUninterruptibly(QuorumFutureFactory.getQuorumFuture(futures));
```

I improved the future especially to address this, but instead of returning "unwritten" it is throwing new exception. Let me see whether it is more reasonable to return a special value.